plutus-core/plutus-core/src/PlutusCore/Evaluation/Machine/ExMemoryUsage.hs

```diff
@@ -221,6 +222,11 @@ instance ExMemoryUsage Integer where
     memoryUsage i = singletonRose $ memoryUsageInteger i
     {-# INLINE memoryUsage #-}
 
+instance ExMemoryUsage Natural where
+    -- FIXME: this is an over-approximation. Figure it out later when costing is done.
```

Comment:
If you're going to do this then it should probably be the same as for `Integer`.  It would be a bit weird if the memory usage of `1234567::Natural` and `1234567::Integer` were different.

Reply:
You are right, in hindsight it should be the same since Natural goes via Integer in our case.